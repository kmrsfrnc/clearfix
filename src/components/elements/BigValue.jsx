import React from 'react';
import PropTypes from 'prop-types';

import Clipboardable from './Clipboardable';

import './BigValue.scss';

class BigValue extends React.Component {
  static propTypes = {
    children: PropTypes.string,
    prefix: PropTypes.string,
  };

  static defaultProps = {
    children: '',
    prefix: '',
  };

  render() {
    const { children, prefix } = this.props;

    return (
      <div className="big-value">
        <span className="big-value__prefix">
          {prefix}
        </span>
        <Clipboardable value={children} component="span">
          <strong className="clickable">
            {children}
          </strong>
        </Clipboardable>
      </div>
    );
  }
}

export default BigValue;
