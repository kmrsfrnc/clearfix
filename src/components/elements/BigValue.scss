.big-value {    
  font-size: 60px;

  &__prefix {
    display: block;
    font-size: 14px;
    font-weight: 600;
    font-style: italic;
    color: rgba(0,0,0,.15);
  }
}