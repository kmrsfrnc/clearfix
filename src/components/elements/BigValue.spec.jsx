import React from 'react';
import { shallow } from 'enzyme';

import BigValue from './BigValue';

describe('BigValue', () => {
  it('matches snapshot', () => {
    expect(shallow(<BigValue prefix="mock prefix">test</BigValue>)).toMatchSnapshot();
  });
});
