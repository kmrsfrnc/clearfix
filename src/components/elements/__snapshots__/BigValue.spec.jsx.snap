// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigValue matches snapshot 1`] = `
<div
  className="big-value"
>
  <span
    className="big-value__prefix"
  >
    mock prefix
  </span>
  <Clipboardable
    component="span"
    value="test"
  >
    <strong
      className="clickable"
    >
      test
    </strong>
  </Clipboardable>
</div>
`;
